File: Math/2443. Sum of Number and Its Reverse/sumOfNumberAndItsReverse.js
//        JS
// Time complexity -  O (NlogN)
// Space complexity - O (1)
var sumOfNumberAndReverse = function(num) {
    var reverseNumber = function(n) {
        let reversed_n = 0;
        while (n > 0) {
            reversed_n = reversed_n * 10 + n % 10;
            n = Math.floor(n / 10);
        }
        return reversed_n;
    };

    for (let x = 0; x <= num; ++x) {
        if (x + reverseNumber(x) === num) {
            return true;
        }
    }
    return false;
};

File: java/SumOfNumberAndItsReverse.java

//        Java
// Time complexity -  O (NlogN)
// Space complexity - O (1)
public class SumOfNumberAndItsReverse {

    public boolean sumOfNumberAndReverse(int num) {
        for (int x = 0; x <= num; x++) {
            if (x + reverse(x) == num) {
                return true;
            }
        }
        return false;
    }

    private int reverse(int x) {
        int reversed = 0;
        while (x > 0) {
            reversed = reversed * 10 + x % 10;
            x /= 10;
        }
        return reversed;
    }
}
